### DISCUSSION TITLE
Cloud limit

### DISCUSSION THREAD

**Discussion by LeafiDev** [TurboWarp/packager#278] [Q&A] [answered]
Is there a cloud limit for local storage cause I have a lot of cloud variables and it disabled the cloud option

**Comment by hello-smile6** [comment]
> Is there a cloud limit for local storage cause I have a lot of cloud variables and it disabled the cloud option

I don't think so.
https://github.com/TurboWarp/packager/blob/1cd7949630dab912df91f0b44cf0fbae0f805613/src/p4/PackagerOptions.svelte#L496-L534
Doesn't seem like anything would disable it for localStorage.

**Reply by LeafiDev** [reply]
![image](https://user-images.githubusercontent.com/83046401/150701028-81323667-8a8d-470b-8ee6-356e5c365cd4.png)
![image](https://user-images.githubusercontent.com/83046401/150701040-bb24cac5-2589-4a69-a187-206c89d0b9ff.png)

**Reply by LeafiDev** [reply]
> > Is there a cloud limit for local storage cause I have a lot of cloud variables and it disabled the cloud option
> 
> I don't think so.
> 
> https://github.com/TurboWarp/packager/blob/1cd7949630dab912df91f0b44cf0fbae0f805613/src/p4/PackagerOptions.svelte#L496-L534
> 
> 
> Doesn't seem like anything would disable it for localStorage.

In the actual editor

**Reply by hello-smile6** [reply]
> > > Is there a cloud limit for local storage cause I have a lot of cloud variables and it disabled the cloud option
> > 
> > 
> > I don't think so.
> > https://github.com/TurboWarp/packager/blob/1cd7949630dab912df91f0b44cf0fbae0f805613/src/p4/PackagerOptions.svelte#L496-L534
> > 
> > Doesn't seem like anything would disable it for localStorage.
> 
> In the actual editor

Huh. Maybe try removing any restrictions that you can find.

**Comment by GarboMuffin** [comment, marked as answer]
For compatibility reasons with Scratch, the desktop editor has the same 10 cloud variable limit.

However, if you name your variables so that they start with `☁`, the packager will convert them to cloud variables for you regardless of that limit

**Reply by LeafiDev** [reply]
I've been using scratch for 6 years and I didn't know this ._.
